flycheck-coala.el: json -> format output

fleimgruber opened the pull request:
The ``--format`` option is more robust than the ``--json`` option when
logging output is mixed with the output of ``:command``;
``json-read-from-string`` breaks in this case.

Fixes https://github.com/coala/coala-emacs/issues/15

fleimgruber committed:
flycheck-coala.el: json -> format output

The ``--format`` option is more robust than the ``--json`` option when
logging output is mixed with the output of ``:command``;
``json-read-from-string`` breaks in this case.

Fixes https://github.com/coala/coala-emacs/issues/15

sils commented on flycheck-coala.el:
```diff
@@ -30,42 +30,14 @@
-  :command ("coala" "--json" "--find-config" "--files" source)
-  :error-parser flycheck-coala-parse-json
+  :command ("coala" "--format" "{line}:{column}:{severity}:{message}" "--find-config" "--files" source)
+  :error-patterns  ((error (or "None" line) ":" (or "None" column) ":" (any "03") ":" (message))
```
why 03 and not 3?

sils replied:
if 03 for a reason this needs a comment IMO

fleimgruber replied:
I do not use numerical error codes any more, see below.

userzimmermann commented on flycheck-coala.el:
```diff
@@ -30,42 +30,14 @@
-  :command ("coala" "--json" "--find-config" "--files" source)
-  :error-parser flycheck-coala-parse-json
+  :command ("coala" "--format" "{line}:{column}:{severity}:{message}" "--find-config" "--files" source)
+  :error-patterns  ((error (or "None" line) ":" (or "None" column) ":" (any "03") ":" (message))
```
Can there really be severity `0`?

fleimgruber replied:
I do not use numerical error codes any more, see below.

userzimmermann commented on flycheck-coala.el:
```diff
@@ -30,42 +30,14 @@
-  :error-parser flycheck-coala-parse-json
+  :command ("coala" "--format" "{line}:{column}:{severity}:{message}" "--find-config" "--files" source)
+  :error-patterns  ((error (or "None" line) ":" (or "None" column) ":" (any "03") ":" (message))
+                    (warning (or "None" line) ":" (or "None" column) ":" (any "2") ":" (message))
```
`":2:"` should be enough instead of using `(any "...")`

fleimgruber replied:
dito

userzimmermann reviewed (changes requested):
Hey @fleimgruber :) Thanks for the PR! Always great to see other Emacs enthusiasts in the coala community... I am only wondering about some little things ;)

fleimgruber commented:
Thanks for your comments!  I changed this to use `severity_str` and literal matches since these map nicely to `flycheck` error levels.  I also added tests to this, see #11.  WDYT?

userzimmermann commented:
ack https://github.com/coala/coala-emacs/pull/16/commits/a9253f192403f067c9a62b47897fb51b4a9a3903

userzimmermann commented:
@rultor merge

rultor commented:
> @rultor merge

@userzimmermann OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/11910-306918433)

userzimmermann commented:
@fleimgruber Now create the PR for https://github.com/coala/coala-emacs/issues/11 :)

rultor commented:
> @rultor merge

@userzimmermann Done! FYI, the full log is [here](http://www.rultor.com/t/11910-306918433) (took me 2min)
